core/string/ustring.cpp

```diff
@@ -3524,10 +3524,12 @@ String String::format(const Variant &values, const String &placeholder) const {
 		for (int i = 0; i < values_arr.size(); i++) {
 			String i_as_str = String::num_int64(i);
 
-			if (values_arr[i].get_type() == Variant::ARRAY) { //Array in Array structure [["name","RobotGuy"],[0,"godot"],["strength",9000.91]]
+#ifndef DISABLE_DEPRECATED
+			if (values_arr[i].get_type() == Variant::ARRAY) { // Array in Array structure [["name","RobotGuy"], [0,"godot"], ["strength",9000.91]].
 				Array value_arr = values_arr[i];
 
 				if (value_arr.size() == 2) {
+					WARN_DEPRECATED_MSG("Arrays structured as key-value pairs in String.format() are deprecated. Consider using a Dictionary, instead.");
```

Comment:
`Consider using a Dictionary instead.` sounds better. I don't think the comma is needed here.